app/dao/notifications_dao.py

```diff
@@ -104,11 +113,88 @@ def country_records_delivery(phone_prefix):
     return dlr and dlr.lower() == 'yes'
 
 
+def _get_notification_status_update_statement(notification_id: str, incoming_status: str, **kwargs):
+    """
+    Generates an update statement for the given notification id and status.
+    Preserves status order and ensures a transient status will not override a final status.
+    :param notification_id: String value of notification uuid to be updated
+    :param status: String value of the status the given notification will attempt to update to
+    :return: An update statement to be executed, or None
+    """
+    current_status = Notification.query.filter(Notification.id == notification_id).one().status
```

Comment:
This does not appear to follow sqlalchemy 2.0 query syntax. 
Should be in the form of `select(Notification).where(Notification.id == notification_id)` etc.